kbncontent.go

```diff
@@ -0,0 +1,248 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// Package kbncontent implements routines for analyzing Kibana content.
+//
+// It provides information about Kibana assets and a single source of truth for what things are legacy and should no longer be used.
+package kbncontent
+
+import (
+	"encoding/json"
+
+	// TODO - consider fully replacing jsonpath with objx
+	"github.com/PaesslerAG/jsonpath"
+	"github.com/stretchr/objx"
+)
+
+// Describes a visualization.
+// In this case, "visualization" means anything which can be embedded in a dashboard.
+type VisDesc struct {
+	Doc    map[string]interface{}
+	SoType string
```

Comment:
Nit. In Go it is usual to use abbreviations in short lived variables, but not so much on type declarations. Consider using full names here, this also helps to self-document the type.
```suggestion
type VisualizationDescriptor struct {
	Doc             map[string]interface{}
	SavedObjectType string
```